### ISSUE TITLE
Documentation: hostsblock

### ISSUE THREAD

**Issue by ShalokShalom** [NixOS/nixpkgs#264311]
## Problem

The hostsblock readme explains to enable a provided systemd service. 
I am new to Nix, and fail to understand how this is supposed to work, as the provided command reports that the service is not available. 

Here is the nix expression: https://github.com/NixOS/nixpkgs/blob/nixos-23.05/pkgs/tools/misc/hostsblock/default.nix
And here are the services: https://github.com/gaenserich/hostsblock/tree/master/systemd
The README: https://github.com/gaenserich/hostsblock#archinstall

![Screenshot_20231030_090438](https://github.com/NixOS/nixpkgs/assets/6344099/04523335-abd7-4e97-8709-6dd10dab060c)

## Checklist

- [x] checked [latest Nixpkgs manual] \([source][nixpkgs-source]) and [latest NixOS manual] \([source][nixos-source])
- [x] checked [open documentation issues] for possible duplicates
- [x] checked [open documentation pull requests] for possible solutions

[latest Nixpkgs manual]: https://nixos.org/manual/nixpkgs/unstable/
[latest NixOS manual]: https://nixos.org/manual/nixos/unstable/
[nixpkgs-source]: https://github.com/NixOS/nixpkgs/tree/master/doc
[nixos-source]: https://github.com/NixOS/nixpkgs/tree/master/nixos/doc/manual
[open documentation issues]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+label%3A%229.needs%3A+documentation%22
[open documentation pull requests]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+is%3Apr+label%3A%228.has%3A+documentation%22%2C%226.topic%3A+documentation%22

**Comment by FliegendeWurst** [comment]
Did you add hostsblock to the `systemPackages`?

**Comment by ShalokShalom** [comment]
Nope, I had no idea I am supposed to. I did move it to the `systemPackages` and run nixos-rebuild switch` and it does not find hostsblock either. 

I am used to hblock just working out of the box (on Arch) and that one is still maintained, while hostsblock seems deprecated. 

There, I never needed to configure any systemd services, or anything.

**Comment by eclairevoyant** [comment]
Since there is no hostsblock module you essentially must make one yourself.

<https://nixos.org/manual/nixos/stable/#sect-nixos-systemd-nixos>

>  In order to enable a systemd system service with provided upstream package, use (e.g):
>
> ```nix
> systemd.packages = [ pkgs.packagekit ];
> ```

And this is not pacman on arch, adding packages to `environment.systemPackages` alone is usually insufficient, as packages will never write outside of the store. You have to do some extra steps such as the above.

**Comment by ShalokShalom** [comment]
So writing a package isnt enough, some packages need modules on top, to do stuff like this, yes?
